fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/FilteredAggregationMaker.java

```diff
@@ -6,54 +6,95 @@
 import com.yahoo.bard.webservice.data.metric.LogicalMetric;
 import com.yahoo.bard.webservice.data.metric.MetricDictionary;
 import com.yahoo.bard.webservice.data.metric.TemplateDruidQuery;
-import com.yahoo.bard.webservice.data.metric.mappers.NoOpResultSetMapper;
+import com.yahoo.bard.webservice.druid.model.MetricField;
 import com.yahoo.bard.webservice.druid.model.aggregation.Aggregation;
 import com.yahoo.bard.webservice.druid.model.aggregation.FilteredAggregation;
 import com.yahoo.bard.webservice.druid.model.filter.Filter;
 
+import com.google.common.collect.ImmutableSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.util.Collections;
 import java.util.List;
 
 /**
  * Build a Filtered Aggregation logical metric.
+ * This metric maker takes a metric whose field is an aggregator and builds a metric with the same dependencies except
+ * without the aggregation and with a filtered aggregation instead using a filter fixed at construction.
  */
 public class FilteredAggregationMaker extends MetricMaker {
 
+    protected static final Logger LOG = LoggerFactory.getLogger(FilteredAggregationMaker.class);
+
+    /**
+     * The filter being applied to the metrics created.
+     */
     protected final Filter filter;
+
+    /**
+     * The metric dictionary used to resolve dependent metrics.
+     */
     protected final MetricDictionary metricDictionary;
```

Comment:
Can/should we kick these over to be `private` instead of `protected`? My gut says we should generally not encourage extension, since it makes it harder to change internal variables like these...

/NotABlocker